### ISSUE TITLE
Upgrading types to `@supabase/postgrest-js@1.17.x` raises type errors

### ISSUE THREAD

**Issue by oof2win2** [psteinroe/supabase-cache-helpers#541]
**Describe the bug**
There are type errors when using `@supabase/ssr` instead of `@supabase/supabase-js` when installing all on `@latest`
```
Argument of type 'SupabaseClient<Database, "public", { Tables: { ai_assistant: { Row: { assistant_id: string; color: string; description: string; emoji: string; example_messages: string[]; id: number; image_url: string; premium: boolean; slug: string; tags: string[]; title: string; }; Insert: { ...; }; Update: { ...; }; Relationships...' is not assignable to parameter of type 'SupabaseClient<any, "public", any>'.
  Property 'supabaseUrl' is protected but type 'SupabaseClient<Database, SchemaName, Schema>' is not a class derived from 'SupabaseClient<Database, SchemaName, Schema>'.
```

**To Reproduce**

Install the following dependencies:
```
dependencies:
@supabase-cache-helpers/postgrest-swr 1.10.1
├─┬ @supabase-cache-helpers/postgrest-core 0.8.1
│ ├── @supabase/postgrest-js 1.17.5 peer
│ └─┬ @supabase/supabase-js 2.45.6 peer
│   └── @supabase/postgrest-js 1.16.3
└── @supabase/postgrest-js 1.17.5 peer
@supabase/ssr 0.5.2
└─┬ @supabase/supabase-js 2.45.6 peer
  └── @supabase/postgrest-js 1.16.3
@supabase/supabase-js 2.45.6
└── @supabase/postgrest-js 1.16.3
```
Use the `createBrowserClient` instead of `createClient`

**Expected behavior**
It works normally without type errors. I tried installing latest and matching versions for everything but it just doesn't work.

**Comment by psteinroe** [comment]
thanks for reporting! I did not check what version of `supabase-js` the `ssr` package uses before upgrading to latest, sorry! 

can you try to install `postgrest-swr@1.10.5`?

**Comment by damien-schneider** [comment]
Are you sure this issue is related to @supabase/ssr? I’m currently using @supabase/ssr@0.5.2, and it works well for www.modul.day. However, this seems related to the following issue mentionned in [supabase-cache-helpers #509](https://github.com/psteinroe/supabase-cache-helpers/issues/509#issuecomment-2481542234).

It’s not working effectively for me either with `supabase-js@2.45.6`, which relies on `postgrest-js` versions above `1.17.x`. These versions seem to have different type definitions, potentially causing the problem.

@oof2win2 Could you try with the `"@supabase/supabase-js": "2.46.1"` and `"@supabase/postgrest-js": "1.16.3"`and tell me if you still have the errors to check if the problems comes from there ?

**Comment by oof2win2** [comment]
@psteinroe Updating to `postgrest-swr@1.10.5` sadly did not fix the issue

@damien-schneider with the changed dependencies it works - don't understand how a minor update on `supabase-js` caused a downgrade on `postgrest-js`, that's an issue to raise with the Supabase team though.

Thank you for the help 👍

**Comment by damien-schneider** [comment]
I’m glad it helped! 🙂

However, I believe the issue shouldn’t be closed just yet. It’s not specifically an error with `@supabase/ssr`, but rather a type error introduced by the postgrest-js `1.17.x` upgrade. This issue should still be addressed. Perhaps renaming it to better reflect the type error would provide clearer context.

**Comment by oof2win2** [comment]
I think it's clearer now. Would internally pinning the types to `@supabase/postgrest-js@1` help?

**Comment by damien-schneider** [comment]
> downgrading `@supabase/postgrest-js` to `1.16.3` works for me
> 
> ```
>     "@supabase-cache-helpers/postgrest-swr": "^1.10.5",
>     "@supabase/postgrest-js": "~1.16.3",
>     "@supabase/ssr": "^0.5.2",
>     "@supabase/supabase-js": "^2.47.2",
> ```

Yes this is the solution for now, but we will have to upgrade to new versions of postgrest-js one day

**Comment by psteinroe** [comment]
sorry everyone! I am putting it higher on my list but cannot guarantee any date.

**Comment by psteinroe** [comment]
the main issue is the improved typing of `PostgrestBuilder` which handles `ThrowOnError` now. Unfortunately, I couldn't find a solution for this yesterday. If a typescript wizard wants to take a look, I would be very grateful. just check out #569, and run `pnpm install` and `pnpm run typecheck` and you will see the first errors appear in `postgrest-core`. 

The problem is that I need to be able to pass "any" `PostgrestBuilder`, but with `ThrowOnError` that defaults to `false`, the `value` of `onfulfilled` is always incorrect.

**Comment by psteinroe** [comment]
I *finally* got around to fix this. the latest release is compatible with the latest `supabase-js`  (`2.49.4` at the time of writing).
